### PR TITLE
[BUG] Fixing bug in pickling readonly graphs

### PR DESCRIPTION
**Pull request by BarclayII**
## Description
Currently pickling a readonly graph throws errors.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change

### CODE REVIEW

File: python/dgl/graph_index.py [hunk 1]
```diff
@@ -30,7 +30,8 @@ def __init__(self, handle=None, multigraph=None, readonly=None):
 
     def __del__(self):
         """Free this graph index object."""
-        _CAPI_DGLGraphFree(self._handle)
+        if hasattr(self, '_handle'):
+            _CAPI_DGLGraphFree(self._handle)
```
Comment (jermainewang) [inline python/dgl/graph_index.py]: When will '_handle' not exist?

File: src/graph/graph.cc [hunk 2]
```diff
@@ -516,17 +521,28 @@ std::vector<IdArray> Graph::GetAdj(bool transpose, const std::string &fmt) const
...
     return std::vector<IdArray>{idx, eid};
   } else if (fmt == "csr") {
-    IdArray indptr = IdArray::Empty({static_cast<int64_t>(num_nodes) + 1}, DLDataType{kDLInt, 64, 1},
-                                    DLContext{kDLCPU, 0});
-    IdArray indices = IdArray::Empty({static_cast<int64_t>(num_edges)}, DLDataType{kDLInt, 64, 1}, DLContext{kDLCPU, 0});
-    IdArray eid = IdArray::Empty({static_cast<int64_t>(num_edges)}, DLDataType{kDLInt, 64, 1}, DLContext{kDLCPU, 0});
+    IdArray indptr = IdArray::Empty(
+        {static_cast<int64_t>(num_nodes) + 1},
+        DLDataType{kDLInt, 64, 1},
+        DLContext{kDLCPU, 0});
```
Comment (jermainewang) [inline src/graph/graph.cc]: Our line length setting is 100 characters (>80)

File: tests/compute/test_pickle.py [hunk 3]
```diff
@@ -60,6 +60,8 @@ def test_pickling_frame():
 
 
 def _assert_is_identical(g, g2):
+    assert g.is_multigraph == g2.is_multigraph
+    assert g._graph.is_readonly() == g2._graph.is_readonly()    # TODO: is_readonly interface
```
Comment (jermainewang) [inline tests/compute/test_pickle.py]: TODO?

Comment (BarclayII) [reply tests/compute/test_pickle.py]: I'm thinking of a `g.is_readonly` interface.

File: src/graph/graph.cc (hunk 2 above)
Comment (BarclayII) [reply src/graph/graph.cc]: I assume now this looks OK?

File: python/dgl/graph_index.py (hunk 1 above)
Comment (BarclayII) [reply python/dgl/graph_index.py]: When the graph fails to load from the pickle.

**Pull request merged**